src/test/fs.test.ts

```diff
@@ -0,0 +1,59 @@
+import * as chai from 'chai';
+import iterate from 'iterare';
+import * as fs from 'mz/fs';
+import * as path from 'path';
+import * as rimraf from 'rimraf';
+import * as temp from 'temp';
+import { LocalFileSystem } from '../fs';
+import { path2uri, toUnixPath } from '../util';
+import chaiAsPromised = require('chai-as-promised');
+chai.use(chaiAsPromised);
+const assert = chai.assert;
+
+describe('fs.ts', () => {
+	describe('LocalFileSystem', () => {
+		let temporaryDir: string;
+		let fileSystem: LocalFileSystem;
+		let baseUri: string;
+
+		before(async () => {
+			temporaryDir = await new Promise<string>((resolve, reject) => {
+				temp.mkdir('local-fs', (err: Error, dirPath: string) => err ? reject(err) : resolve(dirPath));
+			});
+			baseUri = path2uri('', temporaryDir) + '/';
+			await fs.mkdir(path.join(temporaryDir, 'foo'));
+			await fs.writeFile(path.join(temporaryDir, 'tweedledee'), 'hi');
+			await fs.writeFile(path.join(temporaryDir, 'tweedledum'), 'bye');
```

Comment:
can you suffix one of the files with a `.ts` file extension to verify the glob star matches files with extensions too?